### ISSUE TITLE
Array bounds error with Controller:MechanicalVentilation and no DesignSpecification:OutdoorAir objects

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#5112]
Helpdesk ticket 10710
User file crashes with empty error file.  Running in debug mode reveals array bounds error [here](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/MixedAir.cc#L3239) in MixedAir.cc 

This should be legal, but the code for Controller:Mechanical is written such that is assumes there will be some DesignSpecification:OutdoorAir objects present.  From the I/O Ref:

_Field: Design Specification Outdoor Air Object Name <x>_
The name of the DesignSpecification:OutdoorAir object, defining the amount of outdoor air, that applies to the zone or zone list. If this field is blank, the corresponding DesignSpecification:OutdoorAir object for the zone will come from the DesignSpecification:OutdoorAir object referenced by the Sizing:Zone object for the same zone. If no such zone match is found, default values from the IDD will be used for the DesignSpecification:OutdoorAir object.

But the getinput code to search the Sizing:Zone objects is commented out [here](https://github.com/NREL/EnergyPlus/blame/develop/src/EnergyPlus/MixedAir.cc#L1675)

But the block above this only validate this field is it is nonblank.

Similar code is commented out for DesignSpecification:ZoneAirDistribution, and it has similar comments in the I/O Ref that imply it can be omitted.
- [ ] Current code implies that Controller:MechanicalVentilation fields "Design Specification Outdoor Air Object Name <x>" and "Design Specification Zone Air Distribution Object Name <x>" are required inputs for every zone in the list.  Need to confirm if this is the intent - but the code has been this way since before v8.2 (did not check back past that).
- [ ] I/O Ref needs to be updated to reflect this.
- [ ] Once this is required, then [this code](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/MixedAir.cc#L3239) should be OK, but may want to add a test anyway? 
- [ ] Or better yet, OAFlowMethod should be stored in the local data structure similar to the other data from DesignSpecificationOutdoorAir, see [here](https://github.com/NREL/EnergyPlus/blame/develop/src/EnergyPlus/MixedAir.cc#L1653-L1656).  
- [ ] If that's done, then also update other places where OAFlowMethod is used.

**Comment by mjwitte** [comment]
@kbenne Note the user indicated that this defect file came from OpenStudio.  May need to add some checks in OpenStudio.

**Comment by kbenne** [comment]
Only skimming here, but the most common issue involves some known OS "quirks". I'm using the polite term here. In any case the Andrew documented these common issues [here](https://unmethours.com/question/1566/severe-error-caused-by-controller-mechanical-ventilation-openstudio-150/). I've never heard about any of those things crashing E+ though. Could be something else entirely going on here. Thanks for pointing it out.

**Comment by mjwitte** [comment]
@kbenne Sort of similar, but different.  The controller:mechanicalventilation object in this case looks like this.

```
Controller:MechanicalVentilation,
  Controller Mechanical Ventilation 1,    !- Name
  Always On Discrete 1,                   !- Availability Schedule Name
  No,                                     !- Demand Controlled Ventilation
  ZoneSum,                                !- System Outdoor Air Method
  ,                                       !- Zone Maximum Outdoor Air Fraction {dimensionless}
  SPACE1-1 Thermal Zone,                  !- Zone Name 1
  ,                                       !- Design Specification Outdoor Air Object Name 1
  SPACE1-1 Thermal Zone Design Spec Zone Air Dist, !- Design Specification Zone Air Distribution Object Name 1
  SPACE2-1 Thermal Zone,                  !- Zone Name 2
  ,                                       !- Design Specification Outdoor Air Object Name 2
  SPACE2-1 Thermal Zone Design Spec Zone Air Dist, !- Design Specification Zone Air Distribution Object Name 2
  SPACE3-1 Thermal Zone,                  !- Zone Name 3
  ,                                       !- Design Specification Outdoor Air Object Name 3
  SPACE3-1 Thermal Zone Design Spec Zone Air Dist, !- Design Specification Zone Air Distribution Object Name 3
  SPACE4-1 Thermal Zone,                  !- Zone Name 4
  ,                                       !- Design Specification Outdoor Air Object Name 4
  SPACE4-1 Thermal Zone Design Spec Zone Air Dist; !- Design Specification Zone Air Distribution Object Name 4
```

**Comment by kbenne** [comment]
@mjwitte any chance there is an osm file available? I'm 90% sure the osm file is missing a designSpecificationOutdoorAir object on the OpenStudio Space associated with the zone attached to the system.

**Comment by mjwitte** [comment]
@kbenne I asked the user for the osm file.  And I also noticed that the use said this was posted to unmethours also as an OpenStudio problem, but I can't find such a question.  So, I asked for that link as well.

**Comment by mjwitte** [comment]
@kbenne This started as unmethours [10090](https://unmethours.com/question/10090/openstudio-designed-airloop-leading-to-energyplus-crash/).  One of the comments indicates OSM files were sent to _aparker_.

**Comment by kbenne** [comment]
@asparke2 ???

**Comment by asparke2** [comment]
@kbenne I don't know where he sent the files, I checked the help email, my work email, and my personal email...probably why I didn't follow up on UH

**Comment by rraustad** [comment]
Anyone working on this? 1 line of code at 3196 MixedAir.

```
if( vent_mech.ZoneDesignSpecOAObjIndex(ZoneIndex) > 0 ) {
int OAFlowMethod = OARequirements( vent_mech.ZoneDesignSpecOAObjIndex( ZoneIndex ) ).OAFlowMethod;
if ( OAFlowMethod == OAFlowPPer || OAFlowMethod == OAFlowSum || OAFlowMethod == OAFlowMax ) {
    TotalPeopleOAFlow += ZoneIntGain( ZoneNum ).NOFOCC * Zone( ZoneNum ).Multiplier * Zone( ZoneNum ).ListMultiplier * vent_mech.ZoneOAPeopleRate( ZoneIndex );
}
}
```

Any reason why mechanical ventilation should not work without specifying number of people?

**Comment by rraustad** [comment]
A warning already exists. Not sure what defaults are used in this case.

```
** Severe  ** Controller:MechanicalVentilation="CONTROLLER MECHANICAL VENTILATION 1", Design Specification Outdoor Air Object Name blank
**   ~~~   ** For Zone="SPACE1-1 THERMAL ZONE".
**   ~~~   ** This field either needs to be filled in in this object or Sizing:Zone object.
**   ~~~   ** For this run, default values for these fields will be used.
```

**Comment by rraustad** [comment]
Looks like there is a default:

![5112_oaperperson](https://cloud.githubusercontent.com/assets/5778156/10009639/11ed8e54-60ae-11e5-8530-3afa062bb114.png)

**Comment by rraustad** [comment]
So much for a default.

![5112_mechventresult](https://cloud.githubusercontent.com/assets/5778156/10009743/4abe3cb4-60af-11e5-8ec6-dabc9aeef0f8.png)

**Comment by rraustad** [comment]
Created a test file where DesignSpecification:OutdoorAir objects uses default value:

```
DesignSpecification:OutdoorAir,
  SZ DSOA SPACE1-1,        !- Name
  sum,                     !- Outdoor Air Method
  0.00944,                 !- Outdoor Air Flow per Person {m3/s-person}
  0.000,                !- Outdoor Air Flow per Zone Floor Area {m3/s-m2}
  0.0;                     !- Outdoor Air Flow per Zone {m3/s}
```

The defect file and this test file show identical results.
Used Windiff to compare eio (only difference is VentMech reported values), csv (no diff's), and html file  (only difference are time stamp).

PR https://github.com/NREL/EnergyPlus/pull/5223

**Comment by Myoldmopar** [comment]
Closed via #5223.
